### PR TITLE
Move dependencies & add pybind11

### PR DESCRIPTION
**Pull request by breznak**
> Things I have done:
    Emptied out /external/
    Removed /bindings/
    Added /include
    Added boost 1.66 into /external/boost
    Added boost headers to /include/boost
    Added zlib 1.2.11 into /external/zlib
    Added yaml-cpp 0.5.3 into /external/yaml-cpp
    Added yaml-cpp headers into /include/yaml-cpp
    Added pybind11 into /external/pybind11
    Added pybind11 headers into /include/pybind11
    Added gtest 1.8 to /external/gtest
    Added gtest headers to /include/gtest

### CODE REVIEW

File: bindings/py/requirements.txt [hunk 1]
```diff
@@ -1,6 +0,0 @@
-#See http://www.pip-installer.org/en/latest/requirements.html for details
-numpy==1.12.1
```
Comment (breznak) [inline bindings/py/requirements.txt]: is this file moved somewhere else? Even if we remove some deps, we still use it to install others? (numpy,..) 
CI files (.travis.yaml, circle.yml, appveyor.yml) depend install from it.

File: bindings/py/src/nupic/bindings/regions/PyRegion.py [hunk 2]
```diff
@@ -1,406 +0,0 @@
...
-  def __iter__(self):
-    return iter(self._d)
-
-  def __len__(self):
-    return len(self._d)
-
-  def __getitem__(self, key):
-    return self._d[key]
-
-class PyRegion(object):
```
Comment (breznak) [inline bindings/py/src/nupic/bindings/regions/PyRegion.py]: this file is still important for NAPI, right?

File: bindings/py/dummy.c [hunk 3]
```diff
@@ -1,9 +0,0 @@
-#include <stdio.h>
```
Comment (breznak) [inline bindings/py/dummy.c]: so I understand the removed `bindings/py/` is replaced with the PyBind functionality? 
You've mentioned replacing the PyRegion with something else..?

File: external/boost/boost/fusion/container/set/detail/value_of_data_impl.hpp [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+==============================================================================*/
+
+#ifndef BOOST_FUSION_CONTAINER_SET_DETAIL_VALUE_OF_DATA_IMPL_HPP
+#define BOOST_FUSION_CONTAINER_SET_DETAIL_VALUE_OF_DATA_IMPL_HPP
+
+#include <boost/fusion/support/config.hpp>
+
+namespace boost { namespace fusion { namespace extension
```
Comment (breznak) [inline external/boost/boost/fusion/container/set/detail/value_of_data_impl.hpp]: file is located in `external/boost/boost/....` is it correct (the double )?

File: bindings/py/dummy.c (hunk 3 above)
Comment (chhenning) [reply bindings/py/dummy.c]: Correct

File: bindings/py/requirements.txt (hunk 1 above)
Comment (chhenning) [reply bindings/py/requirements.txt]: Not sure we need this folder anymore but if CI scripts insist I can bring it back.

File: bindings/py/src/nupic/bindings/regions/PyRegion.py (hunk 2 above)
Comment (chhenning) [reply bindings/py/src/nupic/bindings/regions/PyRegion.py]: yes, you are correct. I'll add the whole folder.

File: external/boost/boost/fusion/container/set/detail/value_of_data_impl.hpp (hunk 4 above)
Comment (chhenning) [reply external/boost/boost/fusion/container/set/detail/value_of_data_impl.hpp]: external/boost/
you only need to build boost.filesystem and boost.system. All boost headers are in include/boost

File: bindings/py/dummy.c (hunk 3 above)
Comment (breznak) [reply bindings/py/dummy.c]: Even with the tests? You could run `REPO_DIR$ py.test bindings/py/`

Comment (chhenning) [reply bindings/py/dummy.c]: I need to update to python 3 first. Hold on

**Comment by breznak** [comment]
Will be replaced with another PR

**Pull request closed without merging**